### PR TITLE
Fourier interpreted grid filter

### PR DESCRIPTION
**Pull request by FlorianPfaff**
Added Fourier interpreted grid filter

### CODE REVIEW

File: lib/distributions/Circle/FIGDistribution.m [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+            end
+            assert(this.enforcePdfNonnegative == f2.enforcePdfNonnegative);
+            assert(numel(this.gridValues) == numel(f2.gridValues));
+            warnStruct = warning('off', 'Normalization:notNormalized');
+            f = FIGDistribution(this.gridValues.*f2.gridValues, this.transformation, this.enforcePdfNonnegative);
+            warning(warnStruct);
+        end
+        
+        function f = convolve(this, f2)
+            % Multiplies two transformed fourier pdfs (returns transformed result)
```
Comment (LukasDrude) [inline lib/distributions/Circle/FIGDistribution.m]: You may want to double check the doc string here. The function says `convolve` while the text says `multiply`.

File: lib/distributions/Circle/FourierDistribution.m [hunk 2]
```diff
@@ -705,6 +705,18 @@
                         warning('Conversion:NoFormula', 'No explicit formula available, using FFT to get transformation');
                         f = FourierDistribution.fromFunction(@distribution.pdf, noOfCoefficients, desiredTransformation);
                     end
+                case 'FIGDistribution'
+                    % This also works if not right number as it will be
```
Comment (LukasDrude) [inline lib/distributions/Circle/FourierDistribution.m]: Can you add what `right number`  means here? Even number?

File: lib/distributions/Circle/FIGDistribution.m (hunk 1 above)
Comment (FlorianPfaff) [reply lib/distributions/Circle/FIGDistribution.m]: Fixed it, thanks!

File: lib/distributions/Circle/FourierDistribution.m (hunk 2 above)
Comment (FlorianPfaff) [reply lib/distributions/Circle/FourierDistribution.m]: Improved the comment to make clear that I meant that the number of grid points does not need to be equal to the number of Fourier coefficients.

File: lib/distributions/Circle/FIGDistribution.m [hunk 3]
```diff
@@ -0,0 +1,300 @@
+classdef FIGDistribution < AbstractCircularDistribution
+    properties
+        gridValues
+        transformation
+        enforcePdfNonnegative
+    end
+    methods
+        function this = FIGDistribution(gridValues, transformation, enforcePdfNonnegative)
+            if nargin<3, enforcePdfNonnegative=true; end
```
Comment (gerhardkurz) [inline lib/distributions/Circle/FIGDistribution.m:9]: Perhaps give a default value for transform.

File: lib/distributions/Circle/FourierDistribution.m [hunk 4]
```diff
@@ -705,6 +705,19 @@
                         warning('Conversion:NoFormula', 'No explicit formula available, using FFT to get transformation');
                         f = FourierDistribution.fromFunction(@distribution.pdf, noOfCoefficients, desiredTransformation);
                     end
+                case 'FIGDistribution'
+                    % This also works if not the number of coefficients is
```
Comment (gerhardkurz) [inline lib/distributions/Circle/FourierDistribution.m:709]: Remove extra not.

File: lib/distributions/Circle/FIGDistribution.m [hunk 5]
```diff
@@ -0,0 +1,300 @@
+classdef FIGDistribution < AbstractCircularDistribution
```
Comment (gerhardkurz) [inline lib/distributions/Circle/FIGDistribution.m:1]: Perhaps add a reference to the paper.

**Review by gerhardkurz — CHANGES_REQUESTED** [review]

**Comment by FlorianPfaff** [comment]
I will do a new PR with the requested changes and even more enhancements.

**Pull request closed without merging**